src/3d/qgs3dutils.cpp

```diff
@@ -952,3 +956,25 @@ QByteArray Qgs3DUtils::removeDefinesFromShaderCode( const QByteArray &shaderCode
 
   return newShaderCode;
 }
+
+int Qgs3DUtils::getOpenGLMaxClipPlanes()
+{
+  int nrPlanes = 6;
+
+  QOffscreenSurface offscreen;
+  QOpenGLContext context;
+
+  offscreen.setFormat( QSurfaceFormat::defaultFormat() );
+  offscreen.create();
```

Comment:
Since this is to be used in the scene's constructor, can't we use its surface from `mEngine->surface()` instead of creating an offscreen surface?

Reply:
+1

I can also imagine that over time, there would be more checks for various limits of GPU/drivers, so it would be good to have them all in one place in Qgs3DMapScene (e.g. constructor could do multiple checks at once with a single context, rather than creating another context for each check)

Reply:
+1

I can also imagine that over time, there would be more checks for various limits of GPU/drivers, so it would be good to have them all in one place in Qgs3DMapScene (e.g. constructor could do multiple checks at once with a single context, rather than creating another context for each check)

Reply:
Done.